### PR TITLE
Allow `ICommandLineOptions` inside the `IPlatformOutputDevice` extension

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
closes https://github.com/microsoft/testfx/issues/3552

### CODE REVIEW

File: src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Testing.Platform.CommandLine;
+
+internal class CommandLineOptionsProxy : ICommandLineOptions
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs]:
```suggestion
internal sealed class CommandLineOptionsProxy : ICommandLineOptions
```

File: src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+    private ICommandLineOptions? _commandLineOptions;
+
+    public bool IsOptionSet(string optionName) => _commandLineOptions is null
+            ? throw new InvalidOperationException("The ICommandLineOptions has not been built yet or is no more usable at this stage.")
+            : _commandLineOptions.IsOptionSet(optionName);
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs]:
```suggestion
    public bool IsOptionSet(string optionName)
        => _commandLineOptions is null
            ? throw new InvalidOperationException("The ICommandLineOptions has not been built yet or is no more usable at this stage.")
            : _commandLineOptions.IsOptionSet(optionName);
```

File: src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Testing.Platform.CommandLine;
+
+internal class CommandLineOptionsProxy : ICommandLineOptions
+{
+    private ICommandLineOptions? _commandLineOptions;
+
+    public bool IsOptionSet(string optionName) => _commandLineOptions is null
+            ? throw new InvalidOperationException("The ICommandLineOptions has not been built yet or is no more usable at this stage.")
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs:14]: We should have this entry in the resx

File: src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+            : _commandLineOptions.IsOptionSet(optionName);
+
+    public bool TryGetOptionArgumentList(string optionName, [NotNullWhen(true)] out string[]? arguments) => _commandLineOptions is null
+            ? throw new InvalidOperationException("The ICommandLineOptions has not been built yet or is no more usable at this stage.")
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs:19]: We should have this entry in the resx

File: src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+            : _commandLineOptions.TryGetOptionArgumentList(optionName, out arguments);
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs]:
```suggestion
    public bool TryGetOptionArgumentList(string optionName, [NotNullWhen(true)] out string[]? arguments)
        => _commandLineOptions is null
            ? throw new InvalidOperationException("The ICommandLineOptions has not been built yet or is no more usable at this stage.")
            : _commandLineOptions.TryGetOptionArgumentList(optionName, out arguments);
```

File: src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs [hunk 6]
```diff
@@ -187,18 +187,61 @@ public async Task<ITestHost> BuildAsync(
...
+        // Set the concrete command line options to the proxy.
+        commandLineOptionsProxy.SetCommandLineOptions(commandLineHandler);
+
+        // Add FileLoggerProvider if needed
+        if (loggingState.FileLoggerProvider is not null)
+        {
+            Logging.AddProvider((_, _) => loggingState.FileLoggerProvider);
+        }
+
+        // Get te command line options
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs]:
```suggestion
        // Get the command line options
```

File: src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs (hunk 3 above)
Comment (MarcoRossignoli) [reply src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs:14]: We miss it also for the MessageBus..will followup

File: src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs (hunk 4 above)
Comment (MarcoRossignoli) [reply src/Platform/Microsoft.Testing.Platform/CommandLine/CommandLineOptionsProxy.cs:19]: ditto

**Pull request merged**